Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/HotKey/HotKeyWidgetRegistrationInterface.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/RTTI/TypeInfoSimple.h>
+#include <AzCore/RTTI/RTTIMacros.h>
+#include <AzCore/std/string/string.h>
+
+class QWidget;
+
+namespace AzToolsFramework
+{
+    //! HotKeyWidgetRegistrationInterface
+    //! Interface to manage hotkeys in the Editor.
+    class HotKeyWidgetRegistrationInterface
+    {
+    public:
+        AZ_RTTI(HotKeyWidgetRegistrationInterface, "{B9197853-9655-4B72-885C-F02B9B63164F}");
+
+        //! Tries assigning a widget to an Action Context.
+        //! If the system has not yet been initialized, the system will store the arguments and assign the widgets
+        //! after the registration hooks have been called.
+        //! @param contextIdentifier The identifier of the action context to assign the widget to.
+        //! @param widget The widget to assign to the action context.
+        virtual void AssignWidgetToActionContext(const AZStd::string& actionContextIdentifier, QWidget* widget) = 0;
```

Comment:
The lack of efficient string parameter usage here is saddening.

These parameters are better of being an `AZStd::string` parameter, where it gets moved into the `HotKeyActionContextPair` at the end or at the very least an `AZStd::string_view` which avoids an unnecessary allocation when passing in a character literal.
The string_view parameter at least supports passing in an the value from an `AZStd::fixed_string` without any need for conversion.

Here is a link to the O3DE C++ Best Practices section on using accepting string parameters in functions: https://github.com/o3de/sig-core/blob/main/governance/C%2B%2B-Best-Practices-Guide.md#string-view-vs-const-string

Reply:
Using an `AZStd::string` as a key within an `unordered_map` is perfectly fine and interoperates well with an `AZStd::string_view`.
Both the `AZStd::string` [hash function](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/string/string.h#L2138-L2146) and the general [AZStd::equal_to](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/functional_basic.h#L137-L146) function that the hash_table class uses are "transparent".

A [transparent class](https://en.cppreference.com/w/cpp/utility/functional/equal_to_void#Notes) allows those operators to be used without converting the type to an `AZStd::string` avoiding any copies.

That being said, you don't need to address this here, however keep in mind going forward that a `const string&` is the worst case function parameter type for accepting a string.
That type can be replaced with either `string`, `string_view` or `const char*` in way that provides benefits in each case without the drawbacks.